pkg/core/candidate/validator.go

```diff
@@ -35,22 +36,45 @@ func (v *validator) republish(b bytes.Buffer) error {
 	return nil
 }
 
-// Make sure the hash is correct, to avoid malicious nodes from
+// Make sure the hash and root are correct, to avoid malicious nodes from
 // overwriting the candidate block for a specific hash
-func checkHash(b bytes.Buffer) error {
-	cm := &Candidate{}
+func checkHashAndRoot(b bytes.Buffer) error {
```

Comment:
rename this `check`. One can read the code to see what checking means. This way we can add or remove checks without leaving inconsistent names around or having to touch multiple functions